packages/core/src/controllers/ChainController.ts

```diff
@@ -519,5 +520,48 @@ export const ChainController = {
         allAccounts: []
       })
     )
+  },
+
+  async disconnect() {
+    try {
+      const disconnectResults = await Promise.allSettled(
```

Comment:
Do we need `Promise.allSettled` ? Maybe `Promise.all` is enough i don't think it'll throw an error since we have try and catch block.